src/parser/scalar.js

```diff
@@ -223,19 +223,7 @@ module.exports = {
             node = this.node("nowdoc");
             value = this.next().text();
             // strip the last line return char
-            const lastCh = value[value.length - 1];
-            if (lastCh === "\n") {
-              if (value[value.length - 2] === "\r") {
-                // windows style
-                value = value.substring(0, value.length - 2);
-              } else {
-                // linux style
-                value = value.substring(0, value.length - 1);
-              }
-            } else if (lastCh === "\r") {
-              // mac style
-              value = value.substring(0, value.length - 1);
-            }
+            value = value.replace(/(\r\n?|\n) *$/, "");
```

Comment:
not 100% sure but I believe that's not only spaces, could be tabs also. In order to avoid extra regex call could you do something like that :

```
if (this.lexer.heredoc_label.indentation > 0) {
  value = value.substring(0, value.length - this.lexer.heredoc_label.indentation;
}
// and keep the rest the same 
const lastCh = value[value.length - 1];
```